### ISSUE TITLE
PA2 Level 6 StackOverflowError

### ISSUE THREAD

**Issue by nigel-tannn** [nus-cs2030/2223-s2#881]
Not sure if it's because of the indexing for the splitting but I checked and I think it's correct?  Not sure why a StackOverflowError occurs when I run Level 6.

This is the splitting function where I take the List<Integer> and return a Pair<List<Integer>, List<Integer>>

![Screenshot 2023-04-16 at 11 30 36 PM](https://user-images.githubusercontent.com/101518315/232323514-30ea31d9-b4c6-44e1-9c79-6ac647a1ed1c.png)

**Comment by Aron113** [comment]
instead of + 1, it should be + intsize() % 2. Take the case whereby the size of the list is 2, if you work it out, the list will not split properly.

**Comment by Aron113** [comment]
instead of + 1, it should be + intsize() % 2. Take the case whereby the size of the list is 5, if you work it out, the list will not split properly.

**Comment by Aron113** [comment]
sorry, correction to the above. Maybe a better way to put the endpoints is:
 (intList.size() / 2) + ( intList.size() % 2)

**Comment by Danpng01** [comment]
I'm thinking its due to the fact that your bounds are not neccesarliy integers, so you can either typecast or use conditional statements (even / odd list size) to create both lists

**Comment by edmundnwl** [comment]
the bounds seem right to me. but instead of intList shouldn't they be list as it was the argument of the lambda expression?

**Comment by vancwe03** [comment]
you can try storing the (intList.size() + 1) / 2 in an int variable

**Comment by nigel-tannn** [comment]
> the bounds seem right to me. but instead of intList shouldn't they be list as it was the argument of the lambda expression?

Update: yeah this was the issue HAHA thanks for this!! Would've been stuck for a long time otherwise.

**Comment by andersooi** [comment]
Yeah the should be list instead of intList, other than that you've pretty much got it.

**Comment by nigel-tannn** [comment]
Ran into this though, any ideas why? 

<img width="799" alt="Screenshot 2023-04-17 at 10 48 37 AM" src="https://user-images.githubusercontent.com/101518315/232366108-cd3b6f8f-4530-4cbe-aa97-df7ee95d5e3f.png">

**Comment by andersooi** [comment]
Hmm, does your SumList only have a single constructor and does not define any other methods?

**Comment by leezhanpeng** [comment]
> Ran into this though, any ideas why?
> 
> <img alt="Screenshot 2023-04-17 at 10 48 37 AM" width="799" src="https://user-images.githubusercontent.com/101518315/232366108-cd3b6f8f-4530-4cbe-aa97-df7ee95d5e3f.png">

@nigel-tannn Make sure you satisfy these criteria:
![image](https://user-images.githubusercontent.com/96569391/232577839-6b35bdee-cced-4c26-b902-e8bb5db723d9.png)

**Comment by justineeyx** [comment]
> Ran into this though, any ideas why?
> 
> <img alt="Screenshot 2023-04-17 at 10 48 37 AM" width="799" src="https://user-images.githubusercontent.com/101518315/232366108-cd3b6f8f-4530-4cbe-aa97-df7ee95d5e3f.png">

im also facing this issue, how did u fix it? i alr made sure i only have one constructor that takes in a list of integers

**Comment by stxe09** [comment]
move your method into your super() if you haven't already. I think you should be able to adapt what you have to something similar to the one in the walkthrough.
